src/nvim/tui/term_input.inl

```diff
@@ -162,54 +162,56 @@ static void timer_cb(uv_timer_t *handle)
 
 static bool handle_bracketed_paste(TermInput *input)
 {
-  char *ptr = rbuffer_read_ptr(input->read_buffer);
-  size_t len = rbuffer_pending(input->read_buffer);
-  if (len > 5 && (!strncmp(ptr, "\x1b[200~", 6)
-        || !strncmp(ptr, "\x1b[201~", 6))) {
-    bool enable = ptr[4] == '0';
-    // Advance past the sequence
-    rbuffer_consumed(input->read_buffer, 6);
-    if (input->paste_enabled == enable) {
-      return true;
-    }
-    if (enable) {
-      // Get the current mode
-      int state = get_real_state();
-      if (state & NORMAL) {
-        // Enter insert mode
-        input_enqueue(cstr_as_string("i"));
-      } else if (state & VISUAL) {
-        // Remove the selected text and enter insert mode
-        input_enqueue(cstr_as_string("c"));
-      } else if (!(state & INSERT)) {
-        // Don't mess with the paste option
+  RBUFFER_UNTIL_READ_OR_EMPTY(input->read_buffer, ptr, len) {
+    if (len > 5 && (!strncmp(ptr, "\x1b[200~", 6)
+          || !strncmp(ptr, "\x1b[201~", 6))) {
```

Comment:
Does not work for the wrap around case:
A suffix of the string `"\x1b[200~"` could be in the second half.